budgetkey_data_pipelines/pipelines/procurement/tenders/exemptions/exemptions_scraper.py

```diff
@@ -0,0 +1,161 @@
+            href = a_elt.attrib.get("href", "")
+            if href.startswith(base_exemption_message_url):
+                yield href
+
+
+    def _get_page_text(self, form_data=None):
+        response = self._session.request("POST" if form_data else "GET",
+                                         timeout=self._timeout,
+                                         url="http://www.mr.gov.il/ExemptionMessage/Pages/SearchExemptionMessages.aspx",
+                                         data=form_data)
+        response.raise_for_status()
+        return response.text
+
+    def _get_page_text_retry(self, form_data=None):
+        i = 0
+        while True:
+            try:
+                return self._get_page_text(form_data)
+            except (HTTPError, ConnectionError) as e:
+                i += 1
+                if i > self._max_retries:
+                    raise TooManyFailuresException("too many failures, last exception message: {}".format(e))
+                else:
+                    logger.exception(e)
+                    time.sleep(self._wait_between_retries)
+
+    def _has_next_page(self):
+        if self._cur_page_num == 0:
+            return True
+        elif 0 < self._max_pages <= self._cur_page_num:
+            return False
+        else:
+            num_pages = self._get_num_pages()
+            return self._cur_page_num < num_pages
+
+    def _get_num_pages(self):
+        records_range_str = self._page(".resultsSummaryDiv").text()
+        # "tozaot 1-10 mitoch 100 reshumot
+        if len(records_range_str.split(' ')) == 3:  # lo nimtzeu reshoomot
+            # results_range = [0, 0]
+            total_results = 0
+        else:
+            # results_range = [int(x) for x in records_range_str.split(' ')[1].split('-')]
+            total_results = int((records_range_str.split(' ')[3]))
+        records_per_page = 10
+        return math.ceil(total_results / records_per_page)
+
+    def _get_next_page_form_data(self, publisher_id):
+        form_data = {}
+        # copy all the form data from the html input elements
+        for input_elt in self._page("#aspnetForm input"):
+            if input_elt.attrib.get("name") and input_elt.attrib.get("value"):
+                form_data[input_elt.attrib["name"]] = input_elt.attrib["value"]
+        for select_elt in self._page("#WebPartWPQ3 select"):
+            if select_elt.attrib.get("name"):
+                if select_elt.attrib["name"].endswith("$ddlPublisher"):
+                    # set the publisher id
+                    form_data[select_elt.attrib["name"]] = publisher_id
+                elif select_elt.attrib["name"]:
+                    form_data[select_elt.attrib["name"]] = 0
+        for input_elt in self._page("#WebPartWPQ3 input"):
+            if input_elt.get("name"):
+                form_data[input_elt.attrib["name"]] = ""
+        # hard-code the form data with relevant elements from the html input elements
```

Comment:
it's not entirely unused, I'm collecting data from the input fields and then using it below

but, I agree that it's not clear, I'll refactor it a bit